### PR TITLE
Bye bye nanoid, Hello uuid!

### PR DESCRIPTION
**Pull request by bigfoodK**
# What
- Remove nanoid
- Remove Id
- Add uuid

# Why
uuid is king and god.
- Because it supports Copy
- Because we can make uuid in compile time
- Because it supports serde

### CODE REVIEW

File: namui/src/namui/random/nano_id.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#[cfg(target_family = "wasm")]
```
Comment (bigfoodK) [inline namui/src/namui/random/nano_id.rs:1]: Just moved from mod.rs

**Comment by namse** [comment]
@bigfoodK When did you use try_from for Id? I guess that's not a good idea to write/read Id in string in hexadecimal because it will be too long.

**Comment by namse** [comment]
base64 or 16 byte id instead 512 bit.

**Comment by namse** [comment]
Anyway, Display is not for data serialization. Use serde or make it shorter and use raw buffer.

**Review by namse — CHANGES_REQUESTED** [review]
> Use serde or make it shorter and use raw buffer

Request changes. Review rejected.

**Comment by bigfoodK** [comment]
base64 looks good. I will try that

**Comment by namse** [comment]
@bigfoodK Change it 12 byte or 16 bytes, it seems too long for identification. uuid even use 16 bytes.

**Comment by namse** [comment]
OK Sure, Let me know when you are ready to get review

**Comment by namse** [comment]
@bigfoodK is it still needed to use TryFrom?

**Comment by bigfoodK** [comment]
@namse Yes. For creating non-random Id from out of namui

**Comment by namse** [comment]
@bigfoodK Do they exchange id information using string? why not just u8 slice?

**Comment by bigfoodK** [comment]
@namse No. but developer need to create non-random Id for testing or level design.
Using a single line of base64 string would be more readable and editable than a u8 slice.

**Comment by namse** [comment]
I'm confused as to what is the difference between nanoid and nanoid. Should I deprecate the nanoid side?

**Comment by namse** [comment]
What I expected to Id is Copy impl. To make it possible, I used fixed sized u8 array. So I'm confused because you are trying to make it string.

**Comment by bigfoodK** [comment]
I will not exchange Ids in state of String.
But I will use str to creating non-random Id.
ex)
```rust
let pre_assigned_id_for_test: Id = Id::from("ZdNAI2qYjIKZ/030AbSxGw");
```

File: namui/src/namui/random/id.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+}
+impl std::fmt::Display for Id {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // print in hexadecimal
+        for i in 0..self.values.len() {
+            write!(f, "{:02x}", self.values[i])?;
+        }
+        Ok(())
+    }
+}
```
Comment (namse) [inline namui/src/namui/random/id.rs:25]: Please update here with below string type.

File: namui-animation-editor/src/time_point_editor/line_edit_window/render/mod.rs [hunk 3]
```diff
@@ -37,32 +37,47 @@ impl LineEditWindow {
         wh: Wh<Px>,
         layer: &Layer,
         line: ImageInterpolation,
-        point_id: &str,
+        point_id: Uuid,
     ) -> RenderingTree {
         let layer_id = layer.id.clone();
-        let point_id = point_id.to_string();
+        let id_interpolations = ImageInterpolation::iter()
```
Comment (namse) [inline namui-animation-editor/src/time_point_editor/line_edit_window/render/mod.rs:43]: I hate below code, will fix dropdown props next time.

File: namui/src/namui/common/text/multiline_caret.rs [hunk 4]
```diff
@@ -162,7 +162,7 @@ impl MultilineCaret<'_> {
                 .get_glyph_widths(glyph_ids, Some(self.line_texts.paint));
 
             for glyph_width in glyph_widths.into_iter() {
-                let last = caret_positions.last().unwrap();
+                let last = caret_positions.last().unwrap().clone();
```
Comment (namse) [inline namui/src/namui/common/text/multiline_caret.rs:165]: Error since new rust version.

File: namui/src/namui/skia/web/canvas_kit/mod.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-#![allow(unused_doc_comments)]
+#![allow(unused_doc_comments, dead_code)]
```
Comment (namse) [inline namui/src/namui/skia/web/canvas_kit/mod.rs:1]: It shows previous PR merged code.
I made ticket about this.
https://support.github.com/ticket/personal/0/1807703

**Comment by namse** [comment]
@bigfoodK Make a review please. then I will make a final confirm.

File: luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_edit_modal/render/character.rs [hunk 6]
```diff
@@ -24,13 +24,13 @@ impl CharacterEditModal {
...
                     WithAddButton::Item(character) => {
-                        let is_selected = self.character_id.as_deref() == Some(character.id());
+                        let is_selected = self.character_id == Some(character.id());
 
                         let (is_character_cell_in_editing_text_mode, editing_text) =
                             if let Some(EditingTextMode::CharacterName { character_id, text }) =
                                 &self.editing_text_mode
                             {
-                                (character_id == character.id(), Some(text))
+                                (*character_id == character.id(), Some(text))
```
[the remark this answers is not in the document]
Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/loaded/components/character_edit_modal/render/character.rs:33]: It's good, but it's not necessary.

File: namui/src/namui/random/uuid.rs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    Uuid::from_u128(hasher.finish128().into())
+}
+
+#[cfg(test)]
+mod tests {
+    use wasm_bindgen_test::wasm_bindgen_test;
+
+    #[test]
+    #[wasm_bindgen_test]
+    fn try_from_base64_str() {
```
Comment (bigfoodK) [inline namui/src/namui/random/uuid.rs:20]: Maybe we should change test name.

**Review by namse — APPROVED** [review]

**Pull request merged**